core/domain/exp_domain_test.py

```diff
@@ -3516,5 +3616,60 @@ def test_that_mapping_is_correct_for_series_of_changes(self):
         }
         self.assertEqual(
             new_mapping.exploration_version, new_exploration.version)
-        self.assertEqual(new_mapping.state_names_to_ids, expected_mapping)
+        self.assertDictEqual(new_mapping.state_names_to_ids, expected_mapping)
         self.assertEqual(new_mapping.largest_state_id_used, 5)
+
+    def test_for_explorations_having_old_states_schema_version(self):
```

Comment:
There seem to be three different behaviors being tested here; can you split them into three different test methods: one for each behavior being verified? That may make it clearer to readers what specifically is wrong if the test ever fails.

Reply:
Done